test/functional/helpers.lua

```diff
@@ -112,6 +112,10 @@ function module.set_session(s)
   session = s
 end
 
+--- This nasty piece of shit does some sneaky stuff under the hood. So basically each use of this
+--- forces a new loop iteration, so you cannot use this to replace an `exec_lua` with multiple
+--- vim.api calls. Die.
+---
```

Comment:
No, consecutive requests may be accidentally executed in the same loop iteration (unless one of them is `nvim_input`), which is also another source of flakiness for code that assume they are executed in different loop iterations.